addons/product_images/wizard/product_fetch_image_wizard.py

```diff
@@ -0,0 +1,264 @@
+    )
+    nb_product_selected = fields.Integer(string="Number of selected products", readonly=True)
+    nb_products_to_process = fields.Integer(string="Number of products to process", readonly=True)
+    nb_product_unable_to_process = fields.Integer(
+        string="Number of product unprocessable",
+        readonly=True,
+    )
+
+    def action_fetch_image(self):
+        """ Trigger a cron job to fetch image on Google
+
+        The first ten images are immediately fetched to improve the user experience. This way, they
+        can immediately view the images and be assured that the task is running well. Also, if any
+        error occurs, it can be thrown to the user. Then, a cron job is triggered to be run as soon
+        as possible, unless the daily request limit has been reached. In that case, the cron job
+        will be scheduled to run a day later.
+
+        :return: A notification to inform the user that the task is triggered
+        :rtype: dict
+        :raises UserError: If the project is misconfigured on Google's side
+        :raises UserError: If the API Key or Search Engine ID is incorrect
+        """
+        self.products_to_process.image_fetch_pending = True
+        try:
+            self._fetch_image_from_google(self._get_products_to_process(10))
+        except HTTPError as http_error:
+            if http_error.response.status_code == requests.codes.too_many_requests:
+                self.env.ref('product_images.ir_cron_fetch_image')._trigger(
+                    fields.Datetime.now() + timedelta(days=1.0)
+                )
+                return {
+                    'type': 'ir.actions.client',
+                    'tag': 'display_notification',
+                    'params': {
+                        'type': 'warning',
+                        'title': _("Product images"),
+                        'message': _("Custom search quota exceeded. We will start again tomorrow"),
+                        'next': {'type': 'ir.actions.act_window_close'},
+                    }
+                }
+            elif http_error.response.status_code == requests.codes.forbidden:
+                project_url = re.search(
+                    r"(?P<url>https?://[^\s]+)",
+                    http_error.response.json().get('error').get('message'),
+                ).group("url")
+                raise UserError(_(
+                    "Custom Search API is not enabled in your Google project. Enable it by "
+                    "visiting %(url)s then retry. If you enabled this API recently, wait a few "
+                    "minutes for the action to propagate to Google's systems and retry.",
+                    url=project_url
+                ))
+            elif http_error.response.status_code == requests.codes.bad_request:
+                raise UserError(_(
+                    "Your API Key or your Search Engine ID is incorrect."
+                ))
```

Comment:
Maybe `_fetch_image_from_google` could just return the response object so the caller can check for these status codes directly instead of having to catch an exception before it can do so? *it* can always `raise_for_status` in the final clause.